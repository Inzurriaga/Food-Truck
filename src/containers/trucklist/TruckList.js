import React, { Component } from "react";
import { connect } from "react-redux"
import { NavLink } from "react-router-dom"

export class TruckList extends Component{
    render(){
        const truckList = this.props.trucks.map(truck => {
            return(
                <NavLink key={truck.id} to={`/truck/${truck.id}`}>
                    <div className="truck-card">
                        <h2>{truck.name}</h2>
                    </div>
                </NavLink>
            )
        })
        return(
            <div className="truck-list-container">
                {
                    truckList
                }
            </div>
        )
    }
}

export const mapStateToProps = (state) => ({
    trucks: state.trucks
})

export default connect(mapStateToProps)(TruckList)

