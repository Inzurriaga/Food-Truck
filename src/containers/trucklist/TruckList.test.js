import React from "react";
import { shallow } from "enzyme"
import { TruckList } from "./TruckList"

const mockTrucks = []

describe("TruckList", () => {
    describe("TruckList", () => {
        let wrapper;
        beforeEach(() => {
            wrapper = shallow(
                <TruckList trucks={mockTrucks}/>
            )
        })
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})