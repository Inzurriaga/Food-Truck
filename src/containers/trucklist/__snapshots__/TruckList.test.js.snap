// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruckList TruckList should match snap shot 1`] = `
<div
  className="truck-list-container"
/>
`;
